### PR TITLE
iox-#2251 Fix 'Listener' examples

### LINKED ISSUE #2251: Listener examples need to take all samples in the callback
## Required information

The listener examples need to show that one needs to take all the samples of a subscriber, else there will be the illusion that samples are lost.

This can happen when a publisher produces more than one sample while the listener callback is executed.  If the next callback execution then takes only one sample, the queue starts to pile up until eventually an overflow happens or the producer is blocked.

The examples need to show the correct usage of the listener.

See also https://github.com/eclipse-iceoryx/iceoryx/issues/1541#issuecomment-1768502268

### PR DESCRIPTION
**Pull request by elBoberido**
## Notes for Reviewer

The listener examples need to drain the subscriber queue in the callback

## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Consider a second reviewer for complex new features or larger refactorings
- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #2251

### CODE REVIEW

File: iceoryx_examples/callbacks/README.md [hunk 1]
```diff
@@ -165,16 +165,17 @@ void heartbeatCallback(iox::popo::UserTrigger*)
 }
 ```
 
-The `onSampleReceivedCallback` is more complex. We first acquire the received
-sample and check which subscriber signaled the event by acquiring the subscriber's
+The `onSampleReceivedCallback` is more complex. We first acquires all the received
```
Comment (elfenpiff) [inline iceoryx_examples/callbacks/README.md:168]:
```suggestion
The `onSampleReceivedCallback` is more complex. We first acquire all the received
```

File: iceoryx_examples/callbacks/README.md [hunk 2]
```diff
@@ -165,16 +165,17 @@ void heartbeatCallback(iox::popo::UserTrigger*)
...
+samples and checks which subscriber signaled the event by acquiring the subscriber's
```
Comment (elfenpiff) [inline iceoryx_examples/callbacks/README.md:169]:
```suggestion
samples and check which subscriber signaled the event by acquiring the subscriber's
```

**Review by elfenpiff — APPROVED** [review]
Minor typos but this PR is fine as it is for me.

File: iceoryx_examples/callbacks/README.md (hunk 2 above)
Comment (elBoberido) [reply iceoryx_examples/callbacks/README.md:169]: I shouldn't type in the evening :sweat_smile:

**Pull request merged**